### ISSUE TITLE
Conda error prevents installation

### ISSUE THREAD

**Issue by riam528** [graphdeco-inria/gaussian-splatting#229]
So I tried installing this software with
`conda env create --file environment.yml`

but it returns
```
in _check_cuda_version
          raise RuntimeError(CUDA_MISMATCH_MESSAGE.format(cuda_str_version, torch.version.cuda))
      RuntimeError:
      The detected CUDA version (12.2) mismatches the version that was used to compile
      PyTorch (11.6). Please make sure to use the same CUDA versions.

```

Cuda 12.2 installed on the OS level, but isn't Conda supposed to be completely separate from the OS packages?

Modifying the environment.yml to get the latest cuda (no idea if those other version numbers are ideal, just tried to list recent ones):
```
dependencies:
  - cudatoolkit=12.2
  - plyfile=0.8.1
  - python=3.11.5
  - pip=23.2.1
  - pytorch=2.0.1
  - torchaudio=0.12.1
  - torchvision=0.15.2
  - tqdm
```

only results in 

```
Solving environment: failed

ResolvePackageNotFound: 
  - cudatoolkit=12.2

```

So there seems to be no way to install it without removing the cuda 12.2?
Can somebody please tell me how to solve this?

**Comment by Snosixtyboo** [comment]
Hi,

i recommend not modifying the environment.yml, since adding some cuda version in there may cause pytorch to actually install without CUDA support. What you would need to do is make sure you have some CUDA SDK with compilers 11 installed (just not 11.6) and then edit your ```Path``` variable so that CUDA 11 is the first (or only) cuda install directory that is found. Then you would need to open a new console and start again.

Hth,
Bernhard

**Comment by riam528** [comment]
So you can have several cuda toolkit versions installed, thank you.
But wouldn't modifying the PATH make everything else in the system use the outdated cuda toolkit then? I could surely change it back and forth every time, but that isn't very comfortable.

What modifications is the Readme talking about in the "Modifications" section ("our (limited) experiments suggest that the codebase works just fine inside a more up-to-date environment (Python 3.8, PyTorch 2.0.0, CUDA 12)") if not editing the environment.yml?

**Comment by Snosixtyboo** [comment]
Hi,

you can edit the environment.yml, you just have to be careful, the versions have to be such that pytorch will install with cuda. So if you pick the above versions, things should work out. This is best confirmed after installation by doing ```import torch``` and checking ```torch.cuda.is_available()```

Hth,
Bernhard

**Comment by riam528** [comment]
Great, but Conda can't install cuda 12.2 ("ResolvePackageNotFound"), the latest version that I could find with `conda search cudatoolkit`
 is 11.8.0. How do I make conda to download cuda 12.2 instead?

**Comment by riam528** [comment]
I'm sorry, did I misunderstand something?
Please help.

**Comment by ChenMa2017** [comment]
Hi,
   you should have two cudatoolkits on your machine. One is the full version cuda=11.8 in the system, the other one is the cuda=11.6 in your conda environment. The cudatoolkit in the conda environment gaussian_splatting does not has the nvcc. 
  Although you have the cuda-12.2 installed in your system, you can still install the other cuda=11.8 under **/usr/local**. Another way is to build one conda environment that has the cuda=11.8 installed. You can build one environment called  ENV_WITH_CUDA_11_8 using **conda install -c "nvidia/label/cuda-11.8.0" cuda-toolkit**.
  Before you run **conda env create --file environment.yml**, you should set up the PATH variable, which helps the system find out the nvcc 11.8.
  **export PATH=/usr/local/cuda-11.8/bin:$PATH** # if you install the cuda 11.8 or 
  **export PATH=/your/conda/envs/ENV_WITH_CUDA_11_8/bin:$PATH** # if you use the conda 
  After setting up the PATH, you can use **nvcc -V** to print the Version and **which nvcc** to find out which nvcc is working.
  Then you will see the version of nvcc is 11,8 instead of 12.2. It means the system uses the CUDA 11.8 instead 12.2. So you do need to change the yml file.
  Now you can run **conda env create --file environment.yml**.
